[WORDPRESS-57] Add date filter, Change category and tag filters, Fix pagination for search results

AnnaChiu95 opened the pull request:
**What problem is this solving?**

- Add date filter
- Change category and tag filters to make request to WordPress API instead of filtering results after getting posts. This is to keep pagination up to date according to filtered results.
- Fix bug in search results pagination where the second page onwards displayed the same posts.

**How should this be manually tested?**

Here's a workspace ([annawordpress](https://annawordpress--rmoils.myvtex.com/learn)) where this can be tested. Otherwise, link these changes to a workspace, enable the filtering options in app settings, and start filtering posts by category, tag, and/or date. Do the same after searching for blog posts with a search term.

polishq reviewed (changes requested):
Functionality looks good! Here are a couple UI/UX suggestions for the date filtering:
- use a flexbox to display the two date fields in line with each other
- give the first date field some placeholder text like "Filter by date range"
- try displaying the two buttons in the same line as the date fields as well
- only render the "Clear" button if the filter is currently active
- add a small amount of padding between each of the filter types

AnnaChiu95 commented:
@polishq Including your suggestions, I changed the "Clear" button to use an "X" icon instead and added CSS handles to the `WordpressDateSelect` component.

polishq reviewed (approved):
LGTM, great job!
